File: easy/Matrix-Rotation/Matrix-Rotation.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowMatrix(line);
        }
    }
    
    public static void ShowMatrix(String line){
        String[] cells = line.split(" ");
        int total = cells.length;
        int large = (int)Math.sqrt(total);
        for(int i=0;i<large;i++){
            for(int j=0; j<large;j++){
                System.out.print(cells[large*(large-j-1)+i ] + " ");
            }
        }
        System.out.println();
    }
}

File: easy/Matrix-Rotation/Matrix-Rotation.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowMatrix(line);
        }
    }
    
    static void ShowMatrix(string line){
        string[] cells = line.Split(' ');
        int total = cells.Length;
        int large = (int)Math.Sqrt(total);
        for(int i=0;i<large;i++){
            for(int j=0; j<large;j++){
                Console.Write(cells[large*(large-j-1)+i ] + " ");
            }
        }
        Console.WriteLine();
    }
}
